### PR TITLE
refactor index Field

### PR DESCRIPTION
**Pull request by qingxinhome**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #7555

## What this PR does / why we need it:
Simplify index-related data structures

### CODE REVIEW

**Review by ouyuanning — APPROVED** [review]

**Comment by sukki37** [comment]
Please add bvt test cases

**Review by sukki37 — COMMENTED** [review]
Please add bvt test cases

**Comment by qingxinhome** [comment]
> Please add bvt test cases

ok, no problem

File: pkg/sql/colexec/s3util.go [hunk 1]
```diff
@@ -118,18 +118,18 @@ func NewWriteS3Container(tableDef *plan.TableDef) *WriteS3Container {
...
-					coldef := indexdef.Field.Cols[j]
-					container.nameToNullablity[coldef.Name] = coldef.Default.NullAbility
-				}
-			} else {
-				continue
-			}
-		}
-	}
+	//if tableDef.Indexes != nil {
+	//	for _, indexdef := range tableDef.Indexes {
```
Comment (nnsgmsone) [inline pkg/sql/colexec/s3util.go:122]: Index This is gone?

**Review by JackTan25 — APPROVED** [review]

**Review by heni02 — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Review by sukki37 — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]

**Pull request merged**